File: Array/#747. Largest Number At Least Twice of Others.cpp
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
        if(nums.size()==1) return 0;
        int fir=INT_MIN;
        int sec=INT_MIN;
        int res;
        for(int i=0; i<nums.size(); i++){
           if(nums[i]>fir){
               sec=fir;
               fir=nums[i];
               res=i;
               continue;
           }
            if(nums[i]>sec) sec=nums[i];
        }
        return fir>=2*sec ? res : -1;
       
    }
};

File: Array/#747. Largest Number At Least Twice of Others.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        max_num, second_max_num = 0, 0
        res = 0

        for i in range(len(nums)):
            if nums[i] > max_num:
                second_max_num = max_num
                max_num = nums[i]
                res = i
            elif nums[i] < max_num and nums[i] > second_max_num:
                second_max_num = nums[i]

        return res if max_num >= 2 * second_max_num else -1